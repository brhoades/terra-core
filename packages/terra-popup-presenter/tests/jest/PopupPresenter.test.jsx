import React from 'react';
import PopupDefaultExample from './PopupDefaultExample';
import PopupAllPropsExample from './PopupAllPropsExample';
import PopupBoundedExample from './PopupBoundedExample';

const defaultPresenter = <PopupDefaultExample />;

// Snapshot Tests
it('should render a default component', () => {
  const popup = shallow(defaultPresenter);
  expect(popup).toMatchSnapshot();
});

// Prop Tests
it('should have the class terra-PopupPresenter-element', () => {
  const popup = shallow(<PopupAllPropsExample />);
  expect(popup).toMatchSnapshot();
  // expect(wrapper.prop('className')).toContain('terra-PopupPresenter-element');
});

it('should have the class terra-PopupPresenter-element', () => {
  const popup = shallow(<PopupAllPropsExample />);
  expect(popup).toMatchSnapshot();
  // expect(wrapper.prop('className')).toContain('terra-PopupPresenter-element');
});

it('should have the class terra-PopupPresenter-element', () => {
  const popup = shallow(<PopupBoundedExample />);
  expect(popup).toMatchSnapshot();
  // expect(wrapper.prop('className')).toContain('terra-PopupPresenter-element');
});
