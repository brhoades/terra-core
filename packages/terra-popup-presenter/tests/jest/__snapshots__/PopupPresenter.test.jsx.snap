exports[`test should render a default component 1`] = `
<div>
  <div
    style="height:20px;width:20px;" />
</div>
`;

exports[`test should shallow a default component 1`] = `
<div>
  <PopupPresenter
    classNameArrow={null}
    classNameContent={null}
    classNameOverlay={null}
    contentAttachment="top center"
    contentDimensions="25x 25x"
    isArrowDisplayed={false}
    isHeaderDisabled={false}
    isOpen={true}
    onRequestClose={[Function]}
    targetRef={[Function]}>
    <p>
      this is popup content
    </p>
  </PopupPresenter>
  <div
    style={
      Object {
        "height": "20px",
        "width": "20px",
      }
    } />
</div>
`;
